### PR TITLE
Upgrade dependencies and target frameworks

### PR DESCRIPTION
**Pull request by erikzhang**

### CODE REVIEW

**Comment by shargon** [comment]
Seems that `dotnet-format` is not working on mac/3x ..

**Comment by shargon** [comment]
I opened a issue in dotnet https://github.com/dotnet/format/issues/400

**Comment by erikzhang** [comment]
I have a question: why do we need both linux and osx in travis?

**Comment by shargon** [comment]
We have two quick options:

- Remove osx
- Check dotnet-format only on linux

**Comment by erikzhang** [comment]
I prefer to remove osx directly.

File: .travis.yml [hunk 1]
```diff
@@ -2,16 +2,11 @@ language: csharp
...
-osx_image: xcode9.1
-
 mono: none
-dotnet: 2.2.300
+dotnet: 3.0.100
 
-before_install:
-  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then ulimit -n 2048; fi
 install:
   - dotnet tool install -g dotnet-format --version 4.0.40103 --add-source https://dotnet.myget.org/F/format/api/v3/index.json
```
Comment (erikzhang) [inline .travis.yml]: Why do we use MyGet source?

**Comment by erikzhang** [comment]
I think we should move `dotnet-format` from `before_script` to `script`.

**Comment by shargon** [comment]
If dotnet-format fail, will continue if is it under script?

**Comment by erikzhang** [comment]
> If dotnet-format fail, will continue if is it under script?

I think travis won't continue if dotnet-format fails in `script`.

**Pull request merged**